mujoco_py/compile_mujoco.py

```diff
@@ -0,0 +1,301 @@
+# This is a stand alone file!!
+# Don't not introduce any dependency on mujoco_py.
+
+import sys
+import distutils
+import os
+import shutil
+from os.path import abspath, dirname, exists, join, getmtime
+from shutil import move
+import glob
+
+import numpy as np
+from Cython.Build import cythonize
+from Cython.Distutils.old_build_ext import old_build_ext as build_ext
+from lockfile import LockFile, LockTimeout
+import subprocess
+
+# Ensures that mujoco_py is not imported.
+sys.modules['mujoco_py'] = None
+
+def manually_link_libraries():
+    raise Exception("manually_link_libraries should have been "
+                    "overridden by discover_mujoco from utils.py")
+
+
+def discover_mujoco():
+    raise Exception("discover_mujoco function should have been "
+                    "overridden by discover_mujoco from utils.py")
+
+
+# This way we import functions from version.py, and utils.py without
+# relying on mujoco_py package.
+with open(join(dirname(__file__), "version.py")) as version_file:
+    exec(version_file.read())
+
+
+with open(join(dirname(__file__), "utils.py")) as version_file:
+    exec(version_file.read())
+
+def get_nvidia_lib_dir():
+    exists_nvidia_smi = subprocess.call("type nvidia-smi", shell=True,
+                                        stdout=subprocess.PIPE, stderr=subprocess.PIPE) == 0
+    if not exists_nvidia_smi:
+        return None
+    docker_path = '/usr/local/nvidia/lib64'
+    if exists(docker_path):
+        return docker_path
+    paths = glob.glob('/usr/lib/nvidia-[0-9][0-9][0-9]')
```

Comment:
Is there a way to do this glob differently? I think there are several cases where `/usr/lib/nvidia-[0-9][0-9][0-9]` doesn't exist even when the drivers are installed. One example I think is when running `nvidia-docker`, in which case the nvidia drivers are copied over to the container somewhere else than what this is expecting.

This is kinda unrelated to this PR, but caught my eye when I was going through the pr.